README.md
# smart-theater-pos

# Usage

## Environment variables

| Name                                | Required | Value                         | Purpose                                 |
|-------------------------------------|----------|-------------------------------|-----------------------------------------|
| `NODE_ENV`                          | true     | development, test, production | environment name                        |
| `REDIS_HOST`                        | true     |                               | redis host                              |
| `REDIS_PORT`                        | true     |                               | redis port                              |
| `REDIS_KEY`                         | true     |                               | redis key                               |
| `REDIS_TLS_SERVERNAME`              | true     |                               | redis tls servername                    |
| `STORAGE_URL`                       | true     |                               | storage url                             |
| `API_ENDPOINT`                      | true     |                               | api endpoint                            |
| `AUTHORIZATION_CODE_CLIENT_ID`      | true     |                               | client id oauth2                        |
| `AUTHORIZATION_CODE_CLIENT_SECRET`  | true     |                               | client secret oauth2                    |
| `AUTHORIZATION_CODE_DOMAIN`         | true     |                               | oauth2 server domain                    |
| `WAITER_SERVER_URL`                 | true     |                               | waiter server url                       |
| `GMO_TOKEN_URL`                     | true     |                               | gmo token url                           |
| `DEBUG`                             | false    |                               | debug                                   |

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/).

## Unreleased

### Added

### Changed

### Fixed

### Security

## 4.5.0 - 2021-07-13

### Changed

-   メンバーシップを使用しての注文をトークン利用へ変更

## 4.4.0 - 2021-07-07

### Changed

-   チケット枚数表記サイズ変更

## 4.3.0 - 2021-07-05

### Changed

-   併用決済可能に変更

## 4.2.0 - 2021-07-02

### Added

-   注文検索でメンバーシップ注文に対応

## 4.1.0 - 2021-07-01

### Added

-   メンバーシップ注文追加
-   メンバーシップを使用しての注文追加

## 4.0.0 - 2021-06-29

### Changed

-   購入関連サービス細分化

## 3.22.0 - 2021-06-24

### Added

-   チケットへ枚数表記追加

## 3.21.0 - 2021-06-22

### Fixed

-   入力関連の Component を細分化

## 3.20.0 - 2021-06-18

### Fixed

-   自由席の場合座席説明が出ないよう修正

## 3.19.0 - 2021-06-17

### Changed

-   性別の value、label を設定可能に変更
-   年齢のバリデーションを設定可能に変更
-   追加特性を購入確認、購入完了画面でラベル表示に変更

## 3.18.0 - 2021-06-16

### Changed

-   cinerino/sdk>=3.2.0 で再実装

## 3.17.0 - 2021-06-15

### Changed

-   予約検索 UI 調整
-   エラーモーダルにエラーコードを表示するよう変更

## 3.16.0 - 2021-06-11

### Changed

-   bootstrap-icons へ変更

## 3.15.0 - 2021-06-10

### Changed

-   パッケージアップデート(@cinerino/sdk, bootstrap)

## 3.14.0 - 2021-06-07

### Changed

-   入場で券種表示変更（アドオン、 座席なし対応）

## 3.13.0 - 2021-06-03

### Changed

-   施設コンテンツ検索を分割して検索へ変更

## 3.12.0 - 2021-06-01

### Added

-   購入情報変更を追加

## 3.11.0 - 2021-05-31

### Added

-   入場で時間外の場合警告表示

## 3.10.0 - 2021-05-26

### Changed

-   カート機能で日付変更可能に変更

## 3.9.0 - 2021-05-25

### Changed

-   決済方法選択 UI 変更

## 3.8.0 - 2021-05-20

### Changed

-   アドオン補足説明を表示するよう変更
-   ムビチケ画像を css で変更可能に変更

## 3.7.0 - 2020-05-19

### Added

-   顧客の追加特性を注文へ継承

## 3.6.20 - 2020-05-12

### Changed

-   注文検索条件 UI 調整

### Fixed

-   注文検索の COA イベント指定検索修正

## 3.6.19 - 2020-05-11

### Added

-   顧客検索追加（非表示）
-   顧客指定での購入追加

## 3.6.18 - 2020-04-22

### Changed

-   ルーティングログを非表示へ変更

## 3.6.17 - 2020-04-16

### Changed

-   注文配送を同期へ変更

### Fixed

-   プロジェクト一覧レイアウト修正

## 3.6.16 - 2020-04-14

### Changed

-   施設テンプレートでムビチケを利用可能へ変更

## 3.6.15 - 2020-04-13

### Added

-   アドオンを CSV へ表示

## 3.6.14 - 2020-04-12

### Changed

-   施設テンプレートでアドオンの個数選択可能に変更

## 3.6.13 - 2020-04-08

### Changed

-   アドオンを入場画面へ表示

## 3.6.12 - 2020-04-05

### Changed

-   redisClient 生成変更

## 3.6.11 - 2020-04-02

### Fixed

-   プロフィールカスタム項目修正
-   注文検索レイアウト修正

## 3.6.10 - 2020-03-31

### Changed

-   レイティング区分検索適用

## 3.6.9 - 2020-03-29

### Changed

-   aws デプロイ削除変更
-   注文検索、非推奨パラメータを削除

### Fixed

-   返品メール修正

## 3.6.8 - 2020-03-24

### Changed

-   セッションの期間を環境変数で管理

## 3.6.7 - 2020-03-10

### Changed

-   スケジュールの時間比較をサーバータイムに変更

## 3.6.6 - 2020-03-09

### Fixed

-   イベント指定での注文検索で余分な施設コンテンツを検索しないよう修正

## 3.6.5 - 2020-03-08

### Changed

-   API ページング待ち時間変更 (500ms => 100ms)

## 3.6.4 - 2020-03-05

### Fixed

-   座席指定無し不具合修正

## 3.6.3 - 2020-02-26

### Changed

-   暫定対応での入場廃止

## 3.6.2 - 2020-02-24

### Fixed

-   iOS ダブルタップ拡大修正
-   パスワード保存修正

## 3.6.1 - 2020-02-22

### Fixed

-   Continue with Google 不具合修正

## 3.6.0 - 2020-02-19

### Changed

-   決済方法を動的に変更

## 3.5.18 - 2020-02-17

### Changed

-   PurchaseEventSeatTicketComponent => PurchaseEventTicketComponent

## 3.5.17 - 2020-02-16

### Added

-   環境変数へ GTM_ID、ANALYTICS_ID を追加

## 3.5.16 - 2020-02-15

### Changed

-   PurchaseEventTicketComponent => PurchaseEventScheduleComponent

## 3.5.15 - 2020-02-13

### Fixed

-   イベント指定での注文検索修正(COA 対応)

## 3.5.14 - 2020-02-10

### Changed

-   PurchaseEventScheduleComponent => PurchaseEventDateComponent

## 3.5.13 - 2020-02-09

### Added

-   Epson EPOS でのドロワー連動追加

## 3.5.12 - 2020-02-08

### Added

-   入場ゲート選択追加

### Fixed

-   index.html へのリクエストヘッダー削除 (lastModified,etag)

## 3.5.11 - 2020-02-04

### Added

-   検索タイプ切り替えボタン表示

## 3.5.10 - 2020-02-02

### Added

-   注文検索へイベント指定検索を追加（検索タイプ切り替えボタン非表示）

## 3.5.9 - 2020-02-01

### Changed

-   座席表文字サイズ変更

## 3.5.8 - 2020-01-29

### Changed

-   入場 UI 変更

## 3.5.7 - 2020-01-25

### Changed

-   注文検索をコンポーネント化

## 3.5.6 - 2020-01-18

### Fixed

-   ウェイターエラーハンドリング修正

## 3.5.5 - 2020-01-14

### Changed

-   スケジュール取得をコンテンツ、施設コンテンツ、イベント取得へ変更

## 3.5.4 - 2020-01-12

### Added

-   healthcheck 用 API 追加

### Changed

-   NODE_ENV を廃止
-   build コマンド変更

### Fixed

-   取引確定エラーハンドリング修正

## 3.5.3 - 2020-01-07

### Changed

-   パッケージ更新

### Fixed

-   iOS ダブルタップ修正

## 3.5.2 - 2020-12-30

### Changed

-   印刷時エラーハンドリング変更

## 3.5.1 - 2020-12-29

### Fixed

-   プロジェクト選択ログアウト修正

## 3.5.0 - 2020-12-25

### Changed

-   QR 期限を 21 日へ変更

### Fixed

-   ウエイターエラーハンドリング修正

## 3.4.21 - 2020-12-24

### Changed

-   返金メール変更

## 3.4.20 - 2020-12-22

### Added

-   予約 CSV へパフォーマンス id 追加

### Changed

-   自由席文言を変更
-   アプリ設定にカラーを持つように変更

## 3.4.19 - 2020-12-21

### Added

-   購入完了メールへ開場時間追加
-   新入場方式に暫定入場方式の互換性を追加

### Changed

-   旧式の入場を削除

## 3.4.18 - 2020-12-17

### Changed

-   購入メール文言調整

### Fixed

-   自由席不具合修正

## 3.4.17 - 2020-12-14

### Added

-   決済方法へ追加特性情報表示追加

### Changed

-   自由席 UI 変更

## 3.4.16 - 2020-12-11

### Added

-   予約メモ有無を設定追加
-   印刷へリトライ処理追加

## 3.4.15 - 2020-12-09

### Added

-   注文取引の明示的な中止

## 3.4.14 - 2020-12-07

### Fixed

-   CSV ダウンロード分割検索修正

## 3.4.13 - 2020-12-06

### Changed

-   注文作成を同期的に実行へ変更

### Fixed

-   Epson プリンター調整

## 3.4.12 - 2020-12-02

### Fixed

-   オンライン表示開始日時不具合修正

## 3.4.11 - 2020-11-27

### Changed

-   MG チケット QR コード対応

### Fixed

-   QR コードカメラ読み取り調整

## 3.4.10 - 2020-11-25

### Changed

-   発券:新しいコードへ変更
-   ドロワーを同期処理へ変更

### Fixed

-   入場エラーでイベント変更される不具合修正

## 3.4.9 - 2020-11-20

### Changed

-   入場:新しいコードへ対応（互換性あり）

## 3.4.8 - 2020-11-12

### Changed

-   QR 表示「Token」以外の発券処理をを非同期へ変更
-   「満席」文字色を赤から白へ変更

### Added

-   取引へ端末情報を追加

## 3.4.7 - 2020-11-11

### Added

-   決済方法へムビチケ購入番号表示追加

## 3.4.6 - 2020-11-10

### Changed

-   ドロワーを開くタイミング変更

## 3.4.5 - 2020-10-23

### Changed

-   入場用 QR2 パターンのコードへの対応

## 3.4.4 - 2020-10-20

### Changed

-   QR 生成変更

## 3.4.3 - 2020-10-15

### Added

-   入場方式追加

### Changed

-   PURCHASE_PRE_SCHEDULE_DATE 廃止

## 3.4.2 - 2020-10-07

### Added

-   ドロワー連携追加

## 3.4.1 - 2020-10-05


public/default/js/environment.js

window.environment = {
    production: false,
    APP_TITLE: 'CINERINO-POS',
    APP_PREFIX: 'CI',
    PROJECT_ID: 'cinerino',
    ENTRANCE_SERVER_URL: '',
    INSTRUCTION_URL: '/storage/pdf/sample.pdf',
    ANALYTICS_ID: '',
    GTM_ID: '',
    VIEW_TYPE: 'cinema',
    STORAGE_NAME: 'CINERINO-POS-STATE',
    STORAGE_TYPE: 'localStorage',
    BASE_URL: '/purchase/root',
    LANGUAGE: ['ja', 'en'],
    PROFILE: [
        { key: 'age', value: '', required: false },
        { key: 'address', value: '東京都港区赤坂３－４－４専修赤坂ビル７階', required: false },
        { key: 'email', value: '', required: true, maxLength: 50 },
        { key: 'gender', value: '', required: false },
        { key: 'givenName', value: '', required: true, pattern: /^[ァ-ヶー]+$/, maxLength: 12 },
        { key: 'familyName', value: '', required: true, pattern: /^[ァ-ヶー]+$/, maxLength: 12 },
        { key: 'telephone', value: '', required: true, maxLength: 15, minLength: 9 }
    ],
    PAYMENT_METHOD_TO_USE: ['Cash', 'EMoney', 'CreditCard', 'RegiGrow', 'Others'],
    PAYMENT_METHOD_CUSTOM: [
        { category: 'CUSTOM-Group', name: { ja: 'カスタム[団体]', en: 'CUSTOM[Group]' } },
        { category: 'CUSTOM-QR', name: { ja: 'カスタム[QR連携あり]', en: 'CUSTOM[QR linkage]' }, qrcode: '{{ orderNumber }}={{ price }}' }
    ],
    REGIGROW_QRCODE: '{{ orderNumber }}={{ price }}',
    DISPLAY_TICKETED_SEAT: true,
    HEADER_MENU: true,
    HEADER_MENU_SCOPE: ['purchase', 'inquiry', 'order', 'reservation', 'admission', 'setting', 'auth', 'development', 'instruction'],
    PURCHASE_CART: false,
    PURCHASE_ITEM_MAX_LENGTH: '50',
    PURCHASE_TRANSACTION_TIME: '150',
    PURCHASE_TRANSACTION_TIME_DISPLAY: true,
    PURCHASE_TRANSACTION_IDENTIFIER: [
        { name: 'custom', value: 'test' }
    ],
    PURCHASE_PRE_SCHEDULE_DATE: '3',
    PURCHASE_SCHEDULE_DEFAULT_SELECTED_DATE: '0',
    PURCHASE_SCHEDULE_STATUS_THRESHOLD_VALUE: '30',
    PURCHASE_SCHEDULE_STATUS_THRESHOLD_UNIT: '%',
    PURCHASE_COMPLETE_MAIL_CUSTOM: true,
    PURCHASE_TERMS: true,
    PURCHASE_WARNING: true,
    INQUIRY_CANCEL: true,
    INQUIRY_QRCODE: true,
    INQUIRY_PRINT: true,
    ORDER_CANCEL: true,
    ORDER_CANCEL_MAIL_CUSTOM: true,
    ORDER_QRCODE: true,
    ORDER_PRINT: true,
    ORDER_LINK: [
        { name: { ja: '印刷[A4]', en: 'print[A4]' }, url: 'https://motionpicture.jp', params: [{ key: 'output', value: 'a4' }, { key: 'orderNumber' }] }
    ],
    PRINT_QRCODE_TYPE: 'token',
    PRINT_QRCODE_CUSTOM: '',
    PRINT_LOADING: true
};
